app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java

```diff
@@ -0,0 +1,137 @@
+package com.eveningoutpost.dexdrip.Services;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import retrofit.Call;
+import retrofit.Response;
+import retrofit.http.GET;
+import retrofit.http.Query;
+import retrofit.Retrofit;
+import retrofit.GsonConverterFactory;
+
+import com.squareup.okhttp.logging.HttpLoggingInterceptor;
+import com.squareup.okhttp.OkHttpClient;
+
+
+public class NsRestApiReader {
+
+	class NightscoutBg {
+	    double xDrip_raw; // raw_data
+	    double xDrip_filtered; // filtered_data;
+	    Long date; // timestamp
+	    double sgv; // calculated_bg
+	}
+
+	class NightscoutMbg {
+	    Long date; // timestamp
+	    double mbg; // calculated_bg
+	    double xDrip_slope;
+	    double xDrip_intercept;
+	    double xDrip_estimate_raw_at_time_of_calibration;
+	    double xDrip_slope_confidence;
+	    double xDrip_sensor_confidence;
+	    double xDrip_raw_timestamp;
```

Comment:
All this fields are needed in xDrip viewer mode for a calibration graph.

But the real issue here is that we need to pass more fields than nightscout are currently using.
nightscout have a meter record and a calibration record that basically are (from xDrip perspective) a calibration.

So, there are a few things to consider:
1) which of them we want to expand. I was using the meter, but it might be a better idea to live meter as it is and extend the calibration.

2) we need to pass the extended fields to xDrip. So, maybe we should have 3 functions that will get nightscout only data:
getNsSgv()
getNsMeter()
getNsCalibration()
-- names are not final, just for the discussion --
and than have 2 more functions that will get the xDrip data (with all the fields in them):
getXdripSgv()
getXdripCalibration()

This will have all the fields that xDrip needs.
In the future, if more fields are needed, we can simply extend the appropriate class and they will be added automatically.

What do you think?